### PR TITLE
slight botany clipping rework

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
removed rng Sampled and instead its fully health dependent

## Why / Balance
1. how it was before you could be unlucky and only get 1 seed or get 6, leading to botanists turning into machines to get a lot of seeds with shift+e/b because of rng kicking in early
2. since you cant get trolled by Sampled anymore you also dont need to make a giant stock of seeds before mutating, you can just spread them out as normal and then clip after mutating knowing that you will be able to get a seed. this was why youd need a giant stock in the first place so now its just a thing you do if the plant is healthy enough or you actually need more seeds

Sampled bool just served to annoy botanists and didnt prevent anything

sampling a plant to death usually gives 3-6 seeds (bungo) things that have more health will give more as you would expect

## Technical details
removed Sampled entirely

also changed a 30/40/50 damage to be 30-50 to be a little less predictable

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
removed the `Sampled` field

**Changelog**
:cl:
- tweak: Clipping plants is now purely based on health instead of a hidden random variable.

### CODE REVIEW

**Comment by Partmedia** [comment]
I believe `Sampled` exists to prevent infinite seed farming using seed clippers. Are you sure removing it is the right way to go?

With the old RNG, I think it's fair that there's only a 30% chance to prevent further clipping, since the seed output was also RNG. But with the guaranteed seed output, I think clipping should also guarantee no further clipping.

**Comment by GreyMario** [comment]
This isn't the way to make sampling "better". Botany player here, the unrealistic part is that you can take multiple samples of a _freshly planted seed._

Require a mature plant in order to take clippings.

**Comment by Kadeo64** [comment]
FUCKING FINALLY!!! NOW I DON'T HAVE TO SPAM SAMPLE!!

**Comment by K-Dynamic** [comment]
Now we might actually have a use for cryox to heal plants to snip them more

File: Content.Server/Botany/Systems/PlantHolderSystem.cs [hunk 1]
```diff
@@ -268,16 +262,13 @@ private void OnInteractUsing(Entity<PlantHolderComponent> entity, ref InteractUs
             var displayName = Loc.GetString(component.Seed.DisplayName);
             _popup.PopupCursor(Loc.GetString("plant-holder-component-take-sample-message",
                 ("seedName", displayName)), args.User);
-            component.Health -= (_random.Next(3, 5) * 10);
+            component.Health -= _random.Next(30, 50);
```
Comment (Partmedia) [inline Content.Server/Botany/Systems/PlantHolderSystem.cs:265]: These are not equivalent. The former returns only 30, 40, or 50. This patch returns all of the intermediate values. Why do you want to do this?

Comment (deltanedas) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs:265]: already mentioned in pr description, it makes it less predictable

**Comment by Partmedia** [comment]
I'm not a big fan of the change here, because make clipping a plant a lot more useful compared to the seed extractor, which does require a plant to be mature before you can harvest seeds.

I agree with the suggestion that you should only be able to clip seeds from a mature plant.

**Comment by deltanedas** [comment]
> because make clipping a plant a lot more useful compared to the seed extractor, which does require a plant to be mature before you can harvest seeds.

plant clipping is already far more useful, i never see seed extractor used except for flora anomaly gatfruit, no other reason for a good botanist to use it

this doesnt make clipping any more op just less annoying

**Comment by Partmedia** [comment]
> plant clipping is already far more useful, i never see seed extractor used except for flora anomaly gatfruit, no other reason for a good botanist to use it
> 
> this doesnt make clipping any more op just less annoying

But that's the problem, I don't feel that clipping should be more useful or more "op", and this just makes it more "op". As suggested above, I think the problem is that clipping needs to be only allowed for mature plants, so I'm hesitant to make it even more stronger right now.

**Comment by deltanedas** [comment]
again this isnt stronger its just less annoying

you can still get infinite seeds with Sampled rng its just annoying for botanists

**Comment by Partmedia** [comment]
Something being annoying makes it more of "debuff", no? I would really prefer for clippers to be more of a "last resort tool" rather than something in common use.

**Comment by EmoGarbage404** [comment]
there's not really a reason to buff clipping to this extent & botany is already more than capable of spamming out seeds. They don't need this buff (which pretty squarely affects plants with long grow times like gatfruit) so i'm opting to close this pr.

**Comment by deltanedas** [comment]
> which pretty squarely affects plants with long grow times like gatfruit

growth time doesnt matter you can clip right after you plant

**Pull request closed without merging**